### PR TITLE
draft/proof-of-concept: frontend trigger SSE handler on lecture creation

### LINKED ISSUE #34: creating lectures
As a VLA member, I love Physik 1 and create this as a lecture type in some form (tbd). The form allows me to enter the name of this lecture. After submitting this form, this lecture is proposed during appointment creation (#27)

### PR DESCRIPTION
**Pull request by JohaJung**
This currently does not work, but I wanted to try whether I get it to work. 

Our TypeScript Pros may continue working on this :)

Once this is ready, it will close #34 

---

### Implementation

- [ ] Does this code change accomplish what it is supposed to do?
- [ ] Can this solution be simplified without unreasonable effort
- [ ] Is the code modular enough?
- [ ] Does similar functionality already exist in the codebase? If yes, why isn’t it reused?
- [ ] Single Responsibility Principle: Does any class serve only one purpose? If not, it should probably split up in multiple classes.
- [ ] Interface Segregation: If interface implemented, is it fully implemented?
- [ ] Dependency Injection: Are central object instantiation processes managed by a single context?
- [ ] If any functionality is missing, is it marked with a comment that starts with TODO:?
- [ ] Do you see any potential to improve the performance of the code significantly? 

### Logic Errors and Bugs

- [ ] Can you think of any use case in which the code does not behave as intended?
- [ ] Can you think of any inputs or external events that could break the code?
- [ ] Are edge cases covered by tests?

### Error Handling and Logging

- [ ] Is error handling done the correct way?
- [ ] Should any logging or debugging information be added or removed?
- [ ] Are error messages user-friendly?

### Dependencies (internal and external)

- [ ] Were updates to documentation, configuration, or readme files made as required by this change?
- [ ] Is the API.md up-to-date?
- [ ] Are the enumerations between frontend and backend synchronized?

### Security and Data Privacy

- [ ] Does the code introduce any security vulnerabilities?
- [ ] Is (user) input validated, sanitized, and escaped to prevent security attacks such as cross-site scripting or SQL injection?
- [ ] Is sensitive data like user data handled and stored securely?
- [ ] Is the right encryption used?
- [ ] Is data retrieved from external APIs or libraries checked for security issues?
- [ ] Were any secrets committed?

### Testing and Testability (later)

- [ ] Have automated tests been added, or have related ones been updated to cover the change?
- [ ] Do the existing tests reasonably cover the code change?
- [ ] Are there some test cases, input or edge cases that should be tested in addition?

### Usability and Accessibility

- [ ] Is the proposed solution (UI) accessible (e.g. colorblindness, ...)?
- [ ] Is the API/UI intuitive to use?
- [ ] Was the app tested on another device?

### Readability

- [ ] Is the code easy to understand?
- [ ] Which parts were confusing to you and why?
- [ ] Can the readability of the code be improved by smaller methods/different names?
- [ ] Is the code located in the right file/folder?
- [ ] Do you think certain methods should be restructured to have a more intuitive control flow?
- [ ] Are there redundant or outdated comments?
- [ ] Could some comments convey the message better?
- [ ] Could some comments be removed by making the code itself more readable?

### CODE REVIEW

File: VLA-Frontend/src/components/sse/SseHandler.ts [hunk 1]
```diff
@@ -104,4 +106,9 @@ export class SSEHandler {
   private static handleDebugEvent(e: MessageEvent){
     alert(e.data);
   }
+
+  private static handleLectureCreatedEvent(e: MessageEvent){
+    // TODO: trigger this handler (or implement otherwise)
+    // useLectures().handleNewLecture(e.data);
```
Comment (JohaJung) [inline VLA-Frontend/src/components/sse/SseHandler.ts]: This is the point where I gave up.

I wanted to tell the useLectures() object/thread that it should fetch the new entity from the API, and add it to its list.

File: VLA-Frontend/src/hooks/useLectures.ts [hunk 2]
```diff
@@ -32,11 +32,22 @@ export function useLectures() {
...
       console.log("response from adding lecture:");
       res.json().then(data => {
+        console.log(data);
+        // setLectures done in separate function, to be triggered by SSE on all clients
+      });
+    });
+  }
+
+  function handleNewLecture(lectureId: LectureId) {
+    fetch(API_URL + "/" + lectureId).then(res => {
```
Comment (JohaJung) [inline VLA-Frontend/src/hooks/useLectures.ts]: Ideally, it is possible to abstract this away for general entities, i.e. the SSE event just passes an entity URL of the backend REST API, and then the frontend automatically detects which entity it has to update.
